src/lib/automaton.c

```diff
@@ -501,7 +501,8 @@ int inputctx_add_cflow(automaton* a, const char* seq, triefunc fxn){
 
 // multiple input escapes might map to the same input
 int inputctx_add_input_escape(automaton* a, const char* esc, uint32_t special,
-                              unsigned shift, unsigned ctrl, unsigned alt){
+                              unsigned shift, unsigned ctrl, unsigned alt,
```

Comment:
dump these arguments. synthesize them from `modifiers` as a single source of truth until we can rid ourselves of them.